src/app/components/navigation/ErrorClear.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import { clearError } from "../../state";
+
+// This component clears the error state when the location changes (i.e. when the user navigates to a new page)
+
+const ErrorClear = () => {
+  const location = useLocation();
+  const [previousLocation, setPreviousLocation] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (previousLocation !== location.pathname) {
+      clearError();
+      setPreviousLocation(location.pathname);
+    }
+  }, [location, previousLocation]);
```

Comment:
I think this code could be simplified even further. It may not work, but it's worth a try: in theory, if you get rid of `previousLocation` and leave `location` as the only dependency, this effect will still run every time `location` changes. If that does not work, you could also try setting `location.pathname` as the dependency instead.

Reply:
Just to clarify: I meant that I think that you can even get rid of `previousLocation` completely, not only in the dependency array.

Reply:
I changed it to depend on location.pathname, so that it only runs when the pathname changes, otherwise it seems to get called more than necessary